cli/runner.go

```diff
@@ -0,0 +1,296 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+)
+
+const ContainerNetName = "dyo-cli"
+
+const OSwindows = "windows"
+const ExecPodman = "podman"
+const ExecPodmanCompose = "podman-compose"
+
+const ExecDocker = "docker"
+const ExecDockerCompose = "docker-compose"
+const ExecWinDocker = "docker.exe"
+const ExecWinDockerCompose = "docker-compose.exe"
+
+// Start compose file
+func RunContainers(start, quiet bool) error {
+	// search for podman/docker
+	executable, err := FindExec(true)
+	if err != nil {
+		return err
+	}
+
+	var cmd *exec.Cmd
+	if start {
+		cmd = exec.Command(executable, "up", "-d")
+	} else {
+		cmd = exec.Command(executable, "down")
+	}
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	log.Println("Executing", executable, "in the background...")
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	// if we want "silent" output we exit early
+	if quiet {
+		return nil
+	}
+
+	// otherwise see logs
+	log.Println("Printing logs:")
+	cmd = exec.Command(executable, "logs", "-f")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Find a compose or docker executable, prioritizing podman
+func FindExec(compose bool) (executable string, errormsg error) {
+	// I refuse to spend more time to debug this OS's blasphemy
+	if runtime.GOOS == OSwindows {
+		if compose {
+			return ExecWinDockerCompose, nil
+		} else {
+			return ExecWinDocker, nil
+		}
+	}
+
+	executables, err := exploreOSPaths()
+	if err != nil {
+		return "", err
+	}
+
+	dockerComposePresent := false
+	dockerPresent := false
+	for _, file := range executables {
+		switch file {
+		case ExecPodmanCompose:
+			if compose {
+				return ExecPodmanCompose, nil
+			}
+		case ExecDockerCompose:
+			dockerComposePresent = true
+		case ExecPodman:
+			if !compose {
+				return ExecPodman, nil
+			}
+		case ExecDocker:
+			dockerPresent = true
+		}
+	}
+
+	if compose {
+		if dockerComposePresent {
+			return ExecDockerCompose, nil
+		}
+	} else {
+		if dockerPresent {
+			return ExecDocker, nil
+		}
+	}
+
+	return "", errors.New("executable not found")
```

Comment:
Error doesn't contain enough info about the executable we are looking for, bad error message, insert more info.

Reply:
done